scripts/api-workflow.py

```diff
@@ -0,0 +1,188 @@
+'''script to run during the devfile/api workflows'''
+import argparse
+import re
+import os
+import typing
+import shutil
+
+def get_version_breakdown(version: str, version_type: str) -> typing.Tuple[int, int, int]:
+  '''get the version breakdown'''
+  major, minor, bug_fix = re.search(r'^(\d+)\.(\d+)\.(\d+)', version).groups()
+
+  if not (isinstance(major, str) and isinstance(minor, str) and isinstance(bug_fix, str)):
+    raise TypeError(f'{version_type} could not be read: {version}')
+
+  return int(major), int(minor), int(bug_fix)
+
+def update_stable_versions(release: bool, version: str) -> typing.List[str]:
+  '''update stable-versions.txt'''
+  # Parse the schema version
+  major, minor, bug_fix = get_version_breakdown(version, 'Inputted version')
+
+  cwd = os.getcwd()
+  stable_versions: typing.List[str] = []
+
+  with open(f'{cwd}/libs/docs/src/config/stable-versions.txt', encoding='UTF-8') as f:
+    for line in f.readlines():
+      line = line.strip()
+      line_major, line_minor, line_bug_fix = get_version_breakdown(line, 'stable-version')
+
+      # Cannot change a stable version unless its a new release
+      if release and (line_major is major) and (line_minor is minor):
```

Comment:
Could lines 31 to 45 be simplified if we just stored the most recent stable version, and if we did something like this:

```
if line_major <= major and line_minor <= minor and line_bug_fix <= bug_fix
    stable_version = version
    write file
```